kubemarine/admission.py

```diff
@@ -86,14 +86,21 @@ def enrich_inventory_pss(inventory: dict, _: KubernetesCluster) -> dict:
             verify_version(item, inventory["rbac"]["pss"]["defaults"][item], minor_version)
     enabled_admissions = inventory["services"]["kubeadm"]["apiServer"]["extraArgs"].get("feature-gates")
     # add extraArgs to kube-apiserver config
-    if enabled_admissions:
-        if 'PodSecurity=true' not in enabled_admissions:
+    if minor_version > 27:
+        if enabled_admissions:
+            inventory["services"]["kubeadm"]["apiServer"]["extraArgs"]["feature-gates"] = enabled_admissions
```

Comment:
Have you checked what happens during upgrade? Will there be old flag remained, or it will be automatically removed? Also, please refactor this code a bit: 1) it have very many duplicated code; 2) for v1.28 it just does nothing with feature-gates

Reply:
The update to 1.28.0 was successful, the flag is being removed